_posts/2021-10-25-g2p-applications.md

```diff
@@ -0,0 +1,79 @@
+- [Writing mappings on your computer]({{ "g2p-basic-mappings-local" | absolute_url }})
+- [Advanced mappings]({{ "g2p-advanced-mappings" | absolute_url }})
+- ReadAlong Studio & Other Applications
+- [Preprocessing mappings]({{ "g2p-preprocess" | absolute_url }})
+- [Contributing]({{ "g2p-contributing" | absolute_url }})
+
+This post will discuss the use of `g2p` in the [ReadAlongs](#readalongs) project for creating interactive audio/text documents.
+
+## Other Applications
+
+There are number of different software tools that are already making use of `g2p`. For general purpose use in Python, have a look at [this post]({{ "g2p-advanced-mappings" | absolute_url }}).
+
+### Convertextract
+
+Convertextract is a tool that lets you use `g2p` mappings to convert text inside a variety of different documents, including Microsoft Office documents like Powerpoint and Word coduments. Have a look at the [convertextract repo readme](https://github.com/roedoejet/convertextract) or at Fineen Davis' [blog post]({{ "convertextract-app" | absolute_url }}) about the Convertextract GUI.
+
+### ReadAlongs
+
+ReadAlongs is a research project from the [National Research Council's Indigenous Language Technology Project](https://nrc.canada.ca/en/research-development/research-collaboration/programs/canadian-indigenous-languages-technology-project).
+
+Language revitalization communities have potentially many different recordings of language and some associated text, but mobilizing these materials into something that is educationally useful can be time consuming. It's also tough from the learner's perspective to try and follow along the text (maybe a Word Document) just by simply playing the audio. It's easy to get lost in the recording quickly, and from personal experience, trying to do that usually involves a lot of frustrating rewinding.
+
+What if we could develop a tool that *automatically* figured out what parts of the audio file corresponded to what parts of the text? In [Natural Language Processing](https://en.wikipedia.org/wiki/Natural_language_processing), this is called *forced alignment*, and this is fundamentally what the ReadAlongs project does. But how does it do it? In part, by using a whole bunch of `g2p` mappings!
+
+First of all, it's good to know that forced alignment is essentially a "solved problem" for languages with a lot of data, thanks to people like David Huggins-Daines who is a collaborator on the ReadAlongs project. And, if you know Python, and have sentence-aligned parallel audio/text data in your language, you can train a model using one of the few tools out there developed for this task, like [the Montreal Forced Aligner](https://montreal-forced-aligner.readthedocs.io/en/latest/). But what if you don't have that much data (or any) and what if you don't know Python? If your language has a `g2p` mapping between its writing system and the ipa, you can likely circumvent that whole process. Here's how:
```

Comment:
few tools -> several tools ?
not a big difference, but "several" highlights that such tools exist, while I feel "few" seems to insist on the fact that there aren't many, which is not the point here.

Reply:
also, capitalize "ipa" in the last sentence of this paragraph.